hazelcast/src/main/java/com/hazelcast/ringbuffer/Ringbuffer.java

```diff
@@ -0,0 +1,259 @@
+ * item in the ringbuffer is overwritten.
+ *
+ * The ringbuffer has 2 always incrementing sequences:
+ * <ol>
+ * <li>
+ * tailSequence: this is the side where the youngest item is found. So the tail is the side of the ringbuffer where
+ * items are added to.
+ * </li>
+ * <li>
+ * headSequence: this is the side where the oldest items are found. So the head is the side where items gets
+ * discarded.
+ * </li>
+ * </ol>
+ * The items in the ringbuffer can be found by a sequence that is in between (inclusive) the head and tail sequence.
+ *
+ * If data is read from a ringbuffer with a sequence that is smaller than the headSequence, it means that the data
+ * is not available anymore and a {@link StaleSequenceException} is thrown.
+ *
+ * A Ringbuffer currently is not a distributed data-structure. So all data is stored in a single partition; comparable to the
+ * IQueue implementation. But we'll provide an option to partition the data in the near future.
+ *
+ * A Ringbuffer can be used in a similar way as a queue, but one of the key differences is that a queue.take is destructive,
+ * meaning that only 1 thread is able to do take an item. A ringbuffer.read is not destructive, so you can have multiple threads
+ * reading the same data.
+ *
+ * The Ringbuffer is the backing data-structure for the reliable {@link com.hazelcast.core.ITopic} implementation. See
+ * {@link com.hazelcast.config.ReliableTopicConfig}.
+ *
+ * @param <E>
+ */
+@Beta
+public interface Ringbuffer<E> extends DistributedObject {
+
+    /**
+     * Returns the capacity of this Ringbuffer.
+     *
+     * @return the capacity.
+     */
+    long capacity();
+
+    /**
+     * Returns the size (number of items) in the ringbuffer.
+     *
+     * If no retention period is set, the size will always be equal to capacity after the head completed the first loop
+     * around the ring. This is because no items are getting retired.
+     *
+     * @return the size.
+     */
+    long size();
+
+    /**
+     * Returns the sequence of the tail. The tail is the side of the ringbuffer where the items are added to.
+     *
+     * The initial value of the tail is -1.
+     *
+     * @return the sequence of the tail.
+     */
+    long tailSequence();
+
+    /**
+     * Returns the sequence of the head. The head is the side of the ringbuffer where the oldest items in the
+     * ringbuffer are found.
+     *
+     * If the RingBuffer is empty, the head will be one more than the tail.
+     *
+     * The initial value of the head is 0 (1 more than tail).
+     *
+     * @return the sequence of the head.
+     */
+    long headSequence();
+
+    /**
+     * Returns the remaining capacity of the ringbuffer.
+     *
+     * The returned value could be stale as soon as it is returned.
+     *
+     * If retention is not set, the remaining capacity will always be the capacity.
+     *
+     * @return the remaining capacity.
+     */
+    long remainingCapacity();
```

Comment:
then I think you should at least change the `config.getCapacity()` to return `long`... 
```java
public long getCapacity() {
      return capacity;
}
```